spec/graphql/authorization_spec.rb

```diff
@@ -72,18 +72,25 @@ class BaseEnum < GraphQL::Schema::Enum
     module HiddenInterface
       include BaseInterface
 
-      def self.visible?(ctx)
-        super && !ctx[:hide]
-      end
+      definition_methods do
+        def visible?(ctx)
+          super && !ctx[:hide]
+        end
 
-      def self.resolve_type(obj, ctx)
-        HiddenObject
+        def resolve_type(obj, ctx)
+          HiddenObject
+        end
       end
     end
 
     module HiddenDefaultInterface
-      include BaseInterface
-      # visible? will call the super method
+      if GraphQL::Schema.use_schema_subset?
+        include HiddenInterface
+      else
+        # Warden will detect no possible types
+        include BaseInterface
+      end
```

Comment:
This is one of the biggest differences between `Warden` and `Subset`: 

- `Warden` would check if an interface (or union) had any visible possible_types; if it didn't, it would hide that type. (This caused some confusion...) 
- `Subset` won't do that: it will allow the query to use the interface (or union) but fail later if `possible_types` is called for that type (because there won't be any possible types to resolve the current object to).